src/main/java/org/apache/sysds/runtime/instructions/spark/SPInstruction.java

```diff
@@ -41,16 +41,15 @@
 	}
 
 	protected final SPType _sptype;
-	protected final Operator _optr;
 	protected final boolean _requiresLabelUpdate;
 
 	protected SPInstruction(SPType type, String opcode, String istr) {
 		this(type, null, opcode, istr);
 	}
 
 	protected SPInstruction(SPType type, Operator op, String opcode, String istr) {
+		super(op);
```

Comment:
can you explain to me this super class call?

Reply:
It calls the super class constructor, which sets the Operator field (which is in the super class).

Reply:
Because they all have the same field, but it was written in each of the classes. I placed it into their super class (Instruction) and then decided that calling the super constructor is a good idea, since we would then be able to change the way the operator is set for all subclasses by changing the Instruction constructor.